### PR TITLE
[RomApp] Removing Online ANN-ROM's dependency on TensorFlow

### PR DESCRIPTION
**Pull request by NicolasSR**
**📝 Description**
The main contribution of this PR is to remove the dependency the online ANN-ROM simulation to TensorFlow, as the NN forward pass and gradient calculations are already implemented in C++ within the solver.

Until now, TF was just being used to get the network layers and pass them to the solver, and to do a forward pass for proper snapshot initialization. Form now on, the network layer weights are already saved in a .npy fomat (list of arrays) once the NN is trained (note that the h5 weights file from TF is still saved aswell for its possible use). So, in the online simulation, that .npy file is retrieved and the contents passed directly to the solver. Then, the forward pass for initialization is done with the implementation in Kratos.

This makes the online ANN-ROM simulation more universal, in case one might want to train their NNs via Pytorch or other modules. It also allows us to upload small weight files within the ROM tests, which was not possible with the .h5 format, as it was ignored by git.

**🆕 Changelog**

- Modified RomNeuralNetworkTrainer class to save weights in .npy format as well as the original .h5.
- Modified ROM_NN_Interface class to make it independent of TensorFlow. Moved it to new file rom_nn_interface.py where TF is not imported.
- Modified RomAnalysis class to use Kratos' own decoder forward pass instead of using TF.
- Modified RomManager to remove check for TF import before using the ROM_NN_Interface.
- Added public method RunDecoder() within the AnnPromGlobalROMBuilderAndSolver class. Then, added it accessible via Python.
- Changed h5 weight files to npy files from the test_structural_rom.npy and test_structural_lspg_rom.npy tests. Removed skip if TF not found for the online-ANN tests.

### CODE REVIEW

File: applications/RomApplication/tests/test_structural_rom.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 from KratosMultiphysics.RomApplication.rom_manager import RomManager
 from pathlib import Path
 import json
+import h5py
```
Comment (Rbravo555) [inline applications/RomApplication/tests/test_structural_rom.py]: Not required anymore, right?

File: applications/RomApplication/tests/test_structural_lspg_rom.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from KratosMultiphysics.RomApplication.rom_manager import RomManager
 from pathlib import Path
 import json
-
+import h5py
```
Comment (Rbravo555) [inline applications/RomApplication/tests/test_structural_lspg_rom.py]: Not required anymore, right?

File: applications/RomApplication/tests/test_structural_rom.py (hunk 1 above)
Comment (NicolasSR) [reply applications/RomApplication/tests/test_structural_rom.py]: Sure! Already taken care of

File: applications/RomApplication/tests/test_structural_lspg_rom.py (hunk 2 above)
Comment (NicolasSR) [reply applications/RomApplication/tests/test_structural_lspg_rom.py]: Same

**Review by Rbravo555 — APPROVED** [review]
LGTM
